### PR TITLE
Added an optimized bubble sort in sorts

### PR DESCRIPTION
**Pull request by rohleet**
#### Description of Change

Added an Optimized Bubble Sort implementation in the sorting section.
This version improves efficiency over the standard Bubble Sort by introducing a swap flag:

During each pass, adjacent elements are compared and swapped if out of order.

A flag is set whenever a swap occurs.

If no swaps happen in a pass, the algorithm terminates early since the array is already sorted.

Key Points:

Best Case (already sorted): O(n)

Average / Worst Case: O(n²)

Space Complexity: O(1) (in-place)

Stable Sort: Yes

This optimization reduces the number of passes in average and best cases, making it more efficient than the naive approach.

#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: sorting/bubble_sort_optimized.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+int main() {
+    vector<int> v={2,8,1,6,2,0,3,6};
+    v=bubble_sort(v);
+    for(int i=0;i<v.size();i++){
+        cout<<v.at(i)<<endl;
+    }
+    return 0;
+}
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:37]:
```suggestion
}

```

File: sorting/bubble_sort_optimized.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+
+/*This version of bubble sort will reduce the number of passes needed if average cases and also the number od swaps needed in each pass which optimize the algorithm*/
+
+
+#include <iostream>
+#include <vector>
+using namespace std;
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:7]:
```suggestion
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]
Please follow the documentation guidelines, The algorithm is good, you need to add documentation, remove the using namespace line.

thank you for raising this pr

File: sorting/bubble_sort_optimized.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#include <iostream>  /// for IO operations
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:1]: You are missing a file header documentation

File: sorting/bubble_sort_optimized.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        std::vector<int> v2 = {5,4,3,2,1};
+        std::vector<int> expected2 = {1,2,3,4,5};
+        assert(bubble_sort(v2) == expected2);
+
+        std::vector<int> v3 = {1,2,3,4,5};
+        std::vector<int> expected3 = {1,2,3,4,5};
+        assert(bubble_sort(v3) == expected3);
+
+        std::cout << "All tests have successfully passed!\n";
+    }
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:51]: This should be outside this namespace

File: sorting/bubble_sort_optimized.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+namespace sorting {
+
+    /**
+     * @brief Sorts the given vector using the bubble sort algorithm.
+     * Optimized by reducing the number of passes in average cases.
+     * @param nums the vector of integers to be sorted
+     * @return reference to the sorted vector
+     */
+    std::vector<int>& bubble_sort(std::vector<int>& nums) {
+        int flag=0;
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:14]: Shouldn't flag be a bool?
```suggestion
        bool flag =  false;
```

File: sorting/bubble_sort_optimized.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        int temp;
+        for(int i=0; i<nums.size()-1;i++){
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:16]: Please lint this file using clang-format and clang tidy

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: sorting/bubble_sort_optimized.cpp [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  const std::vector<int> expected1 = {0, 1, 2, 2, 3, 6, 6, 8};
+  assert(sorting::bubble_sort(v1) == expected1);
+
+  std::vector<int> v2 = {5, 4, 3, 2, 1};
+  const std::vector<int> expected2 = {1, 2, 3, 4, 5};
+  assert(sorting::bubble_sort(v2) == expected2);
+
+  std::vector<int> v3 = {1, 2, 3, 4, 5};
+  const std::vector<int> expected3 = {1, 2, 3, 4, 5};
+  assert(sorting::bubble_sort(v3) == expected3);
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:77]: Add tests cases for negative numbers, empty array.

File: sorting/bubble_sort_optimized.cpp [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * @brief Sorts the given vector using the bubble sort algorithm.
+ *
+ * This optimized bubble sort reduces the number of passes in average cases
+ * by checking if the array is already sorted during each pass.
+ *
+ * @param nums the vector of integers to be sorted
+ * @return reference to the sorted vector
+ */
+std::vector<int>& bubble_sort(std::vector<int>& nums) {
```
Comment (realstealthninja) [inline sorting/bubble_sort_optimized.cpp:40]: Could you try to make it generic by using templates to accommodate comparable data types?

**Review by realstealthninja — CHANGES_REQUESTED** [review]
